mainline @ 159510a is failing CI

Failing run: https://dev.azure.com/numba/numba/_build/results?buildId=6831&view=logs&j=a506c6d0-8428-5bb5-7d36-c242a5467a36&t=3b81a997-a0fe-5621-27e6-18b6744841b9&l=9361

Details: https://gist.github.com/stuartarchibald/ace0bea10f04abb44d65a0ed9f8ba2cb

Dumping out the error:
```
AssertionError: 'cannot index array' not found in 'Failed in nopython mode pipeline (step: nopython frontend)
No implementation of function Function(<function flipud at 0x7f64ef745b00>) found for signature:
 
 >>> flipud(int64)
 
There are 2 candidate implementations:
  - Of which 1 did not match due to:
  Overload in function \'np_flip_ud\': File: numba/np/arrayobj.py: Line 4917.
    With argument(s): \'(int64)\':
   Rejected as the implementation raised a specific error:
     TypingError: Failed in nopython mode pipeline (step: nopython frontend)
   Internal error at <numba.core.typeinfer.StaticGetItemConstraint object at 0x7f64cd53e4d0>.
   \'TestOverloadInlining\' object has no attribute \'DummyType\'
   During: typing of intrinsic-call at /home/vsts/work/1/s/numba/np/arrayobj.py (4930)
   During: typing of static-get-item at /home/vsts/work/1/s/numba/np/arrayobj.py (4930)
   Enable logging at debug level for details.
   
   File "numba/np/arrayobj.py", line 4930:
       def impl(a):
           <source elided>
               raise ValueError(\'Input must be >= 1-d.\')
           return A[::-1, ...]
           ^
   
       Traceback (most recent call last):
         File "/home/vsts/work/1/s/numba/core/typing/templates.py", line 677, in _get_impl
           impl, args = self._impl_cache[cache_key]
       KeyError: (<numba.core.typing.context.Context object at 0x7f64e14c37d0>, (int64,), ())
       During handling of the above exception, another exception occurred:
       Traceback (most recent call last):
         File "/home/vsts/work/1/s/numba/core/typing/templates.py", line 677, in _get_impl
           impl, args = self._impl_cache[cache_key]
       KeyError: (<numba.core.typing.context.Context object at 0x7f64e14c37d0>, (array(int64, 0d, C), Tuple(slice<a:b:c>, ...)), ())
       During handling of the above exception, another exception occurred:
       Traceback (most recent call last):
         File "/home/vsts/work/1/s/numba/core/errors.py", line 738, in new_error_context
           yield
         File "/home/vsts/work/1/s/numba/core/typeinfer.py", line 689, in __call__
           self.resolve(typeinfer, typeinfer.typevars, fnty=fnty)
         File "/home/vsts/work/1/s/numba/core/typeinfer.py", line 586, in resolve
           sig = typeinfer.resolve_call(fnty, pos_args, kw_args)
         File "/home/vsts/work/1/s/numba/core/typeinfer.py", line 1535, in resolve_call
           return self.context.resolve_function_type(fnty, pos_args, kw_args)
         File "/home/vsts/work/1/s/numba/core/typing/context.py", line 207, in resolve_function_type
           res = self._resolve_builtin_function_type(func, args, kws)
         File "/home/vsts/work/1/s/numba/core/typing/context.py", line 224, in _resolve_builtin_function_type
           res = defn.apply(args, kws)
         File "/home/vsts/work/1/s/numba/core/typing/templates.py", line 345, in apply
           sig = generic(args, kws)
         File "/home/vsts/work/1/s/numba/core/typing/templates.py", line 597, in generic
           disp, new_args = self._get_impl(args, kws)
         File "/home/vsts/work/1/s/numba/core/typing/templates.py", line 679, in _get_impl
           impl, args = self._build_impl(cache_key, args, kws)
         File "/home/vsts/work/1/s/numba/core/typing/templates.py", line 710, in _build_impl
           ovf_result = self._overload_func(*args, **kws)
         File "/home/vsts/work/1/s/numba/tests/test_ir_inlining.py", line 660, in overload_dummy_getitem
           if isinstance(obj, self.DummyType):
       AttributeError: \'TestOverloadInlining\' object has no attribute \'DummyType\'
       During handling of the above exception, another exception occurred:
       Traceback (most recent call last):
         File "/home/vsts/work/1/s/numba/core/typeinfer.py", line 154, in propagate
           constraint(typeinfer)
         File "/home/vsts/work/1/s/numba/core/typeinfer.py", line 469, in __call__
           self.fallback(typeinfer)
         File "/home/vsts/work/1/s/numba/core/typeinfer.py", line 689, in __call__
           self.resolve(typeinfer, typeinfer.typevars, fnty=fnty)
         File "/home/vsts/miniconda3/envs/travisci/lib/python3.7/contextlib.py", line 130, in __exit__
           self.gen.throw(type, value, traceback)
         File "/home/vsts/work/1/s/numba/core/errors.py", line 745, in new_error_context
           raise newerr.with_traceback(tb)
         File "/home/vsts/work/1/s/numba/core/errors.py", line 738, in new_error_context
           yield
         File "/home/vsts/work/1/s/numba/core/typeinfer.py", line 689, in __call__
           self.resolve(typeinfer, typeinfer.typevars, fnty=fnty)
         File "/home/vsts/work/1/s/numba/core/typeinfer.py", line 586, in resolve
           sig = typeinfer.resolve_call(fnty, pos_args, kw_args)
         File "/home/vsts/work/1/s/numba/core/typeinfer.py", line 1535, in resolve_call
           return self.context.resolve_function_type(fnty, pos_args, kw_args)
         File "/home/vsts/work/1/s/numba/core/typing/context.py", line 207, in resolve_function_type
           res = self._resolve_builtin_function_type(func, args, kws)
         File "/home/vsts/work/1/s/numba/core/typing/context.py", line 224, in _resolve_builtin_function_type
           res = defn.apply(args, kws)
         File "/home/vsts/work/1/s/numba/core/typing/templates.py", line 345, in apply
           sig = generic(args, kws)
         File "/home/vsts/work/1/s/numba/core/typing/templates.py", line 597, in generic
           disp, new_args = self._get_impl(args, kws)
         File "/home/vsts/work/1/s/numba/core/typing/templates.py", line 679, in _get_impl
           impl, args = self._build_impl(cache_key, args, kws)
         File "/home/vsts/work/1/s/numba/core/typing/templates.py", line 710, in _build_impl
           ovf_result = self._overload_func(*args, **kws)
         File "/home/vsts/work/1/s/numba/tests/test_ir_inlining.py", line 660, in overload_dummy_getitem
           if isinstance(obj, self.DummyType):
       numba.core.errors.InternalError: \'TestOverloadInlining\' object has no attribute \'DummyType\'
       During: typing of intrinsic-call at /home/vsts/work/1/s/numba/np/arrayobj.py (4930)
       During: typing of static-get-item at /home/vsts/work/1/s/numba/np/arrayobj.py (4930)
       The above exception was the direct cause of the following exception:
       Traceback (most recent call last):
         File "/home/vsts/work/1/s/numba/core/types/functions.py", line 298, in get_call_type
           sig = temp.apply(nolitargs, nolitkws)
         File "/home/vsts/work/1/s/numba/core/typing/templates.py", line 345, in apply
           sig = generic(args, kws)
         File "/home/vsts/work/1/s/numba/core/typing/templates.py", line 597, in generic
           disp, new_args = self._get_impl(args, kws)
         File "/home/vsts/work/1/s/numba/core/typing/templates.py", line 679, in _get_impl
           impl, args = self._build_impl(cache_key, args, kws)
         File "/home/vsts/work/1/s/numba/core/typing/templates.py", line 740, in _build_impl
           disp_type.get_call_type(self.context, args, kws)
         File "/home/vsts/work/1/s/numba/core/types/functions.py", line 511, in get_call_type
           self.dispatcher.get_call_template(args, kws)
         File "/home/vsts/work/1/s/numba/core/dispatcher.py", line 311, in get_call_template
           self.compile(tuple(args))
         File "/home/vsts/work/1/s/numba/core/compiler_lock.py", line 32, in _acquire_compile_lock
           return func(*args, **kwargs)
         File "/home/vsts/work/1/s/numba/core/dispatcher.py", line 818, in compile
           cres = self._compiler.compile(args, return_type)
         File "/home/vsts/work/1/s/numba/core/dispatcher.py", line 81, in compile
           raise retval
         File "/home/vsts/work/1/s/numba/core/dispatcher.py", line 91, in _compile_cached
           retval = self._compile_core(args, return_type)
         File "/home/vsts/work/1/s/numba/core/dispatcher.py", line 109, in _compile_core
           pipeline_class=self.pipeline_class)
         File "/home/vsts/work/1/s/numba/core/compiler.py", line 602, in compile_extra
           return pipeline.compile_extra(func)
         File "/home/vsts/work/1/s/numba/core/compiler.py", line 352, in compile_extra
           return self._compile_bytecode()
         File "/home/vsts/work/1/s/numba/core/compiler.py", line 414, in _compile_bytecode
           return self._compile_core()
         File "/home/vsts/work/1/s/numba/core/compiler.py", line 394, in _compile_core
           raise e
         File "/home/vsts/work/1/s/numba/core/compiler.py", line 385, in _compile_core
           pm.run(self.state)
         File "/home/vsts/work/1/s/numba/core/compiler_machinery.py", line 339, in run
           raise patched_exception
         File "/home/vsts/work/1/s/numba/core/compiler_machinery.py", line 330, in run
           self._runPass(idx, pass_inst, state)
         File "/home/vsts/work/1/s/numba/core/compiler_lock.py", line 32, in _acquire_compile_lock
           return func(*args, **kwargs)
         File "/home/vsts/work/1/s/numba/core/compiler_machinery.py", line 289, in _runPass
           mutated |= check(pss.run_pass, internal_state)
         File "/home/vsts/work/1/s/numba/core/compiler_machinery.py", line 262, in check
           mangled = func(compiler_state)
         File "/home/vsts/work/1/s/numba/core/typed_passes.py", line 99, in run_pass
           raise_errors=self._raise_errors)
         File "/home/vsts/work/1/s/numba/core/typed_passes.py", line 71, in type_inference_stage
           infer.propagate(raise_errors=raise_errors)
         File "/home/vsts/work/1/s/numba/core/typeinfer.py", line 1071, in propagate
           raise errors[0]
       numba.core.errors.TypingError: Failed in nopython mode pipeline (step: nopython frontend)
       Internal error at <numba.core.typeinfer.StaticGetItemConstraint object at 0x7f64cd53e4d0>.
       \'TestOverloadInlining\' object has no attribute \'DummyType\'
       During: typing of intrinsic-call at /home/vsts/work/1/s/numba/np/arrayobj.py (4930)
       During: typing of static-get-item at /home/vsts/work/1/s/numba/np/arrayobj.py (4930)
       Enable logging at debug level for details.
       File "numba/np/arrayobj.py", line 4930:
           def impl(a):
               <source elided>
                   raise ValueError(\'Input must be >= 1-d.\')
               return A[::-1, ...]
               ^
  raised from /home/vsts/work/1/s/numba/core/typeinfer.py:1071
  - Of which 1 did not match due to:
  Overload in function \'np_flip_ud\': File: numba/np/arrayobj.py: Line 4917.
    With argument(s): \'(int64)\':
   Rejected as the implementation raised a specific error:
     TypingError: Failed in nopython mode pipeline (step: nopython frontend)
   Internal error at <numba.core.typeinfer.StaticGetItemConstraint object at 0x7f64ce4bfb50>.
   \'TestOverloadInlining\' object has no attribute \'DummyType\'
   During: typing of intrinsic-call at /home/vsts/work/1/s/numba/np/arrayobj.py (4930)
   During: typing of static-get-item at /home/vsts/work/1/s/numba/np/arrayobj.py (4930)
   Enable logging at debug level for details.
   
   File "numba/np/arrayobj.py", line 4930:
       def impl(a):
           <source elided>
               raise ValueError(\'Input must be >= 1-d.\')
           return A[::-1, ...]
           ^
   
       Traceback (most recent call last):
         File "/home/vsts/work/1/s/numba/core/typing/templates.py", line 677, in _get_impl
           impl, args = self._impl_cache[cache_key]
       KeyError: (<numba.core.typing.context.Context object at 0x7f64e14c37d0>, (int64,), ())
       During handling of the above exception, another exception occurred:
       Traceback (most recent call last):
         File "/home/vsts/work/1/s/numba/core/typing/templates.py", line 677, in _get_impl
           impl, args = self._impl_cache[cache_key]
       KeyError: (<numba.core.typing.context.Context object at 0x7f64e14c37d0>, (array(int64, 0d, C), Tuple(slice<a:b:c>, ...)), ())
       During handling of the above exception, another exception occurred:
       Traceback (most recent call last):
         File "/home/vsts/work/1/s/numba/core/errors.py", line 738, in new_error_context
           yield
         File "/home/vsts/work/1/s/numba/core/typeinfer.py", line 689, in __call__
           self.resolve(typeinfer, typeinfer.typevars, fnty=fnty)
         File "/home/vsts/work/1/s/numba/core/typeinfer.py", line 586, in resolve
           sig = typeinfer.resolve_call(fnty, pos_args, kw_args)
         File "/home/vsts/work/1/s/numba/core/typeinfer.py", line 1535, in resolve_call
           return self.context.resolve_function_type(fnty, pos_args, kw_args)
         File "/home/vsts/work/1/s/numba/core/typing/context.py", line 207, in resolve_function_type
           res = self._resolve_builtin_function_type(func, args, kws)
         File "/home/vsts/work/1/s/numba/core/typing/context.py", line 224, in _resolve_builtin_function_type
           res = defn.apply(args, kws)
         File "/home/vsts/work/1/s/numba/core/typing/templates.py", line 345, in apply
           sig = generic(args, kws)
         File "/home/vsts/work/1/s/numba/core/typing/templates.py", line 597, in generic
           disp, new_args = self._get_impl(args, kws)
         File "/home/vsts/work/1/s/numba/core/typing/templates.py", line 679, in _get_impl
           impl, args = self._build_impl(cache_key, args, kws)
         File "/home/vsts/work/1/s/numba/core/typing/templates.py", line 710, in _build_impl
           ovf_result = self._overload_func(*args, **kws)
         File "/home/vsts/work/1/s/numba/tests/test_ir_inlining.py", line 660, in overload_dummy_getitem
           if isinstance(obj, self.DummyType):
       AttributeError: \'TestOverloadInlining\' object has no attribute \'DummyType\'
       During handling of the above exception, another exception occurred:
       Traceback (most recent call last):
         File "/home/vsts/work/1/s/numba/core/typeinfer.py", line 154, in propagate
           constraint(typeinfer)
         File "/home/vsts/work/1/s/numba/core/typeinfer.py", line 469, in __call__
           self.fallback(typeinfer)
         File "/home/vsts/work/1/s/numba/core/typeinfer.py", line 689, in __call__
           self.resolve(typeinfer, typeinfer.typevars, fnty=fnty)
         File "/home/vsts/miniconda3/envs/travisci/lib/python3.7/contextlib.py", line 130, in __exit__
           self.gen.throw(type, value, traceback)
         File "/home/vsts/work/1/s/numba/core/errors.py", line 745, in new_error_context
           raise newerr.with_traceback(tb)
         File "/home/vsts/work/1/s/numba/core/errors.py", line 738, in new_error_context
           yield
         File "/home/vsts/work/1/s/numba/core/typeinfer.py", line 689, in __call__
           self.resolve(typeinfer, typeinfer.typevars, fnty=fnty)
         File "/home/vsts/work/1/s/numba/core/typeinfer.py", line 586, in resolve
           sig = typeinfer.resolve_call(fnty, pos_args, kw_args)
         File "/home/vsts/work/1/s/numba/core/typeinfer.py", line 1535, in resolve_call
           return self.context.resolve_function_type(fnty, pos_args, kw_args)
         File "/home/vsts/work/1/s/numba/core/typing/context.py", line 207, in resolve_function_type
           res = self._resolve_builtin_function_type(func, args, kws)
         File "/home/vsts/work/1/s/numba/core/typing/context.py", line 224, in _resolve_builtin_function_type
           res = defn.apply(args, kws)
         File "/home/vsts/work/1/s/numba/core/typing/templates.py", line 345, in apply
           sig = generic(args, kws)
         File "/home/vsts/work/1/s/numba/core/typing/templates.py", line 597, in generic
           disp, new_args = self._get_impl(args, kws)
         File "/home/vsts/work/1/s/numba/core/typing/templates.py", line 679, in _get_impl
           impl, args = self._build_impl(cache_key, args, kws)
         File "/home/vsts/work/1/s/numba/core/typing/templates.py", line 710, in _build_impl
           ovf_result = self._overload_func(*args, **kws)
         File "/home/vsts/work/1/s/numba/tests/test_ir_inlining.py", line 660, in overload_dummy_getitem
           if isinstance(obj, self.DummyType):
       numba.core.errors.InternalError: \'TestOverloadInlining\' object has no attribute \'DummyType\'
       During: typing of intrinsic-call at /home/vsts/work/1/s/numba/np/arrayobj.py (4930)
       During: typing of static-get-item at /home/vsts/work/1/s/numba/np/arrayobj.py (4930)
       The above exception was the direct cause of the following exception:
       Traceback (most recent call last):
         File "/home/vsts/work/1/s/numba/core/types/functions.py", line 293, in get_call_type
           sig = temp.apply(args, kws)
         File "/home/vsts/work/1/s/numba/core/typing/templates.py", line 345, in apply
           sig = generic(args, kws)
         File "/home/vsts/work/1/s/numba/core/typing/templates.py", line 597, in generic
           disp, new_args = self._get_impl(args, kws)
         File "/home/vsts/work/1/s/numba/core/typing/templates.py", line 679, in _get_impl
           impl, args = self._build_impl(cache_key, args, kws)
         File "/home/vsts/work/1/s/numba/core/typing/templates.py", line 740, in _build_impl
           disp_type.get_call_type(self.context, args, kws)
         File "/home/vsts/work/1/s/numba/core/types/functions.py", line 511, in get_call_type
           self.dispatcher.get_call_template(args, kws)
         File "/home/vsts/work/1/s/numba/core/dispatcher.py", line 311, in get_call_template
           self.compile(tuple(args))
         File "/home/vsts/work/1/s/numba/core/compiler_lock.py", line 32, in _acquire_compile_lock
           return func(*args, **kwargs)
         File "/home/vsts/work/1/s/numba/core/dispatcher.py", line 818, in compile
           cres = self._compiler.compile(args, return_type)
         File "/home/vsts/work/1/s/numba/core/dispatcher.py", line 81, in compile
           raise retval
         File "/home/vsts/work/1/s/numba/core/dispatcher.py", line 91, in _compile_cached
           retval = self._compile_core(args, return_type)
         File "/home/vsts/work/1/s/numba/core/dispatcher.py", line 109, in _compile_core
           pipeline_class=self.pipeline_class)
         File "/home/vsts/work/1/s/numba/core/compiler.py", line 602, in compile_extra
           return pipeline.compile_extra(func)
         File "/home/vsts/work/1/s/numba/core/compiler.py", line 352, in compile_extra
           return self._compile_bytecode()
         File "/home/vsts/work/1/s/numba/core/compiler.py", line 414, in _compile_bytecode
           return self._compile_core()
         File "/home/vsts/work/1/s/numba/core/compiler.py", line 394, in _compile_core
           raise e
         File "/home/vsts/work/1/s/numba/core/compiler.py", line 385, in _compile_core
           pm.run(self.state)
         File "/home/vsts/work/1/s/numba/core/compiler_machinery.py", line 339, in run
           raise patched_exception
         File "/home/vsts/work/1/s/numba/core/compiler_machinery.py", line 330, in run
           self._runPass(idx, pass_inst, state)
         File "/home/vsts/work/1/s/numba/core/compiler_lock.py", line 32, in _acquire_compile_lock
           return func(*args, **kwargs)
         File "/home/vsts/work/1/s/numba/core/compiler_machinery.py", line 289, in _runPass
           mutated |= check(pss.run_pass, internal_state)
         File "/home/vsts/work/1/s/numba/core/compiler_machinery.py", line 262, in check
           mangled = func(compiler_state)
         File "/home/vsts/work/1/s/numba/core/typed_passes.py", line 99, in run_pass
           raise_errors=self._raise_errors)
         File "/home/vsts/work/1/s/numba/core/typed_passes.py", line 71, in type_inference_stage
           infer.propagate(raise_errors=raise_errors)
         File "/home/vsts/work/1/s/numba/core/typeinfer.py", line 1071, in propagate
           raise errors[0]
       numba.core.errors.TypingError: Failed in nopython mode pipeline (step: nopython frontend)
       Internal error at <numba.core.typeinfer.StaticGetItemConstraint object at 0x7f64ce4bfb50>.
       \'TestOverloadInlining\' object has no attribute \'DummyType\'
       During: typing of intrinsic-call at /home/vsts/work/1/s/numba/np/arrayobj.py (4930)
       During: typing of static-get-item at /home/vsts/work/1/s/numba/np/arrayobj.py (4930)
       Enable logging at debug level for details.
       File "numba/np/arrayobj.py", line 4930:
           def impl(a):
               <source elided>
                   raise ValueError(\'Input must be >= 1-d.\')
               return A[::-1, ...]
               ^
  raised from /home/vsts/work/1/s/numba/core/typeinfer.py:1071

During: resolving callee type: Function(<function flipud at 0x7f64ef745b00>)
During: typing of call at /home/vsts/work/1/s/numba/tests/test_np_functions.py (121)


File "numba/tests/test_np_functions.py", line 121:
def flipud(a):
    return np.flipud(a)
    ^

````

--- a/numba/tests/test_ir_inlining.py
+++ b/numba/tests/test_ir_inlining.py
@@ -133,6 +133,8 @@ class InliningBase(TestCase):
         return fir  # for use in further analysis
 
     def make_dummy_type(self):
+        """ Use to generate a dummy type """
+
         # Use test_id to make sure no collision is possible.
         test_id = self.id()
         DummyType = type('DummyTypeFor{}'.format(test_id), (types.Opaque,), {})
@@ -151,8 +153,7 @@ class InliningBase(TestCase):
         def unbox_dummy(typ, obj, c):
             return NativeValue(c.context.get_dummy_value())
 
-        self.Dummy = Dummy
-        self.DummyType = DummyType
+        return Dummy, DummyType
 
 
 # used in _gen_involved
@@ -506,12 +507,6 @@ class TestRegisterJitableInlining(MemoryLeakMixin, InliningBase):
 
 class TestOverloadInlining(MemoryLeakMixin, InliningBase):
 
-    def setUp(self):
-        # only tests for operators need DummyType
-        if 'operator' in self._testMethodName:
-            self.make_dummy_type()
-        super(TestOverloadInlining, self).setUp()
-
     def test_basic_inline_never(self):
         def foo():
             pass
@@ -566,21 +561,22 @@ class TestOverloadInlining(MemoryLeakMixin, InliningBase):
             return +x
 
         dummy_unary_impl = lambda x: True
-        setattr(self.Dummy, '__neg__', dummy_unary_impl)
-        setattr(self.Dummy, '__pos__', dummy_unary_impl)
+        Dummy, DummyType = self.make_dummy_type()
+        setattr(Dummy, '__neg__', dummy_unary_impl)
+        setattr(Dummy, '__pos__', dummy_unary_impl)
 
         @overload(operator.neg, inline='always')
         def overload_dummy_neg(x):
-            if isinstance(x, self.DummyType):
+            if isinstance(x, DummyType):
                 return dummy_unary_impl
 
         @overload(operator.pos, inline='never')
         def overload_dummy_pos(x):
-            if isinstance(x, self.DummyType):
+            if isinstance(x, DummyType):
                 return dummy_unary_impl
 
-        self.check(impl_inline, self.Dummy(), inline_expect={'neg': True})
-        self.check(impl_noinline, self.Dummy(), inline_expect={'pos': False})
+        self.check(impl_inline, Dummy(), inline_expect={'neg': True})
+        self.check(impl_noinline, Dummy(), inline_expect={'pos': False})
 
     def test_inline_operators_binop(self):
 
@@ -590,22 +586,24 @@ class TestOverloadInlining(MemoryLeakMixin, InliningBase):
         def impl_noinline(x):
             return x != 1
 
+        Dummy, DummyType = self.make_dummy_type()
+
         dummy_binop_impl = lambda a, b: True
-        setattr(self.Dummy, '__eq__', dummy_binop_impl)
-        setattr(self.Dummy, '__ne__', dummy_binop_impl)
+        setattr(Dummy, '__eq__', dummy_binop_impl)
+        setattr(Dummy, '__ne__', dummy_binop_impl)
 
         @overload(operator.eq, inline='always')
         def overload_dummy_eq(a, b):
-            if isinstance(a, self.DummyType):
+            if isinstance(a, DummyType):
                 return dummy_binop_impl
 
         @overload(operator.ne, inline='never')
         def overload_dummy_ne(a, b):
-            if isinstance(a, self.DummyType):
+            if isinstance(a, DummyType):
                 return dummy_binop_impl
 
-        self.check(impl_inline, self.Dummy(), inline_expect={'eq': True})
-        self.check(impl_noinline, self.Dummy(), inline_expect={'ne': False})
+        self.check(impl_inline, Dummy(), inline_expect={'eq': True})
+        self.check(impl_noinline, Dummy(), inline_expect={'ne': False})
 
     def test_inline_operators_inplace_binop(self):
 
@@ -615,34 +613,36 @@ class TestOverloadInlining(MemoryLeakMixin, InliningBase):
         def impl_noinline(x):
             x -= 1
 
+        Dummy, DummyType = self.make_dummy_type()
+
         dummy_inplace_binop_impl = lambda a, b: True
-        setattr(self.Dummy, '__iadd__', dummy_inplace_binop_impl)
-        setattr(self.Dummy, '__isub__', dummy_inplace_binop_impl)
+        setattr(Dummy, '__iadd__', dummy_inplace_binop_impl)
+        setattr(Dummy, '__isub__', dummy_inplace_binop_impl)
 
         @overload(operator.iadd, inline='always')
         def overload_dummy_iadd(a, b):
-            if isinstance(a, self.DummyType):
+            if isinstance(a, DummyType):
                 return dummy_inplace_binop_impl
 
         @overload(operator.isub, inline='never')
         def overload_dummy_isub(a, b):
-            if isinstance(a, self.DummyType):
+            if isinstance(a, DummyType):
                 return dummy_inplace_binop_impl
 
         # DummyType is not mutable, so lowering 'inplace_binop' Expr
         # re-uses (requires) copying function definition
         @overload(operator.add, inline='always')
         def overload_dummy_add(a, b):
-            if isinstance(a, self.DummyType):
+            if isinstance(a, DummyType):
                 return dummy_inplace_binop_impl
 
         @overload(operator.sub, inline='never')
         def overload_dummy_sub(a, b):
-            if isinstance(a, self.DummyType):
+            if isinstance(a, DummyType):
                 return dummy_inplace_binop_impl
 
-        self.check(impl_inline, self.Dummy(), inline_expect={'iadd': True})
-        self.check(impl_noinline, self.Dummy(), inline_expect={'isub': False})
+        self.check(impl_inline, Dummy(), inline_expect={'iadd': True})
+        self.check(impl_noinline, Dummy(), inline_expect={'isub': False})
 
     def test_inline_always_operators_getitem(self):
 
@@ -652,19 +652,20 @@ class TestOverloadInlining(MemoryLeakMixin, InliningBase):
         def impl_static_getitem(x):
             return x[1]
 
+        Dummy, DummyType = self.make_dummy_type()
+
         dummy_getitem_impl = lambda obj, idx: None
-        setattr(self.Dummy, '__getitem__', dummy_getitem_impl)
+        setattr(Dummy, '__getitem__', dummy_getitem_impl)
 
         @overload(operator.getitem, inline='always')
         def overload_dummy_getitem(obj, idx):
-            if isinstance(obj, self.DummyType):
+            if isinstance(obj, DummyType):
                 return dummy_getitem_impl
 
         # note getitem and static_getitem Exprs refer to operator.getitem
         # hence they are checked using the same expected key
-        self.check(impl, self.Dummy(), 1, inline_expect={'getitem': True})
-        self.check(impl_static_getitem,
-                   self.Dummy(),
+        self.check(impl, Dummy(), 1, inline_expect={'getitem': True})
+        self.check(impl_static_getitem, Dummy(),
                    inline_expect={'getitem': True})
 
     def test_inline_never_operators_getitem(self):
@@ -675,19 +676,20 @@ class TestOverloadInlining(MemoryLeakMixin, InliningBase):
         def impl_static_getitem(x):
             return x[1]
 
+        Dummy, DummyType = self.make_dummy_type()
+
         dummy_getitem_impl = lambda obj, idx: None
-        setattr(self.Dummy, '__getitem__', dummy_getitem_impl)
+        setattr(Dummy, '__getitem__', dummy_getitem_impl)
 
         @overload(operator.getitem, inline='never')
         def overload_dummy_getitem(obj, idx):
-            if isinstance(obj, self.DummyType):
+            if isinstance(obj, DummyType):
                 return dummy_getitem_impl
 
         # noth getitem and static_getitem Exprs refer to opertor.getitem
         # hence they are checked using the same expect key
-        self.check(impl, self.Dummy(), 1, inline_expect={'getitem': False})
-        self.check(impl_static_getitem,
-                   self.Dummy(),
+        self.check(impl, Dummy(), 1, inline_expect={'getitem': False})
+        self.check(impl_static_getitem, Dummy(),
                    inline_expect={'getitem': False})
 
     def test_inline_stararg_error(self):
@@ -1073,7 +1075,9 @@ class TestOverloadMethsAttrsInlining(InliningBase):
             self.assertTrue(allgetattrs)
 
     def test_overload_method_default_args_always(self):
-        @overload_method(self.DummyType, "inline_method", inline='always')
+        Dummy, DummyType = self.make_dummy_type()
+
+        @overload_method(DummyType, "inline_method", inline='always')
         def _get_inlined_method(obj, val=None, val2=None):
             def get(obj, val=None, val2=None):
                 return ("THIS IS INLINED", val, val2)
@@ -1084,7 +1088,7 @@ class TestOverloadMethsAttrsInlining(InliningBase):
 
         self.check_method(
             test_impl=foo,
-            args=[self.Dummy()],
+            args=[Dummy()],
             expected=(("THIS IS INLINED", 123, None),
                       ("THIS IS INLINED", None, 321)),
             block_count=1,
@@ -1094,7 +1098,9 @@ class TestOverloadMethsAttrsInlining(InliningBase):
         def costmodel(*args):
             return should_inline
 
-        @overload_method(self.DummyType, "inline_method", inline=costmodel)
+        Dummy, DummyType = self.make_dummy_type()
+
+        @overload_method(DummyType, "inline_method", inline=costmodel)
         def _get_inlined_method(obj, val):
             def get(obj, val):
                 return ("THIS IS INLINED!!!", val)
@@ -1105,7 +1111,7 @@ class TestOverloadMethsAttrsInlining(InliningBase):
 
         self.check_method(
             test_impl=foo,
-            args=[self.Dummy()],
+            args=[Dummy()],
             expected=("THIS IS INLINED!!!", 123),
             block_count=1,
             expects_inlined=should_inline,
@@ -1136,7 +1142,9 @@ class TestOverloadMethsAttrsInlining(InliningBase):
         )
 
     def make_overload_attribute_test(self, costmodel, should_inline):
-        @overload_attribute(self.DummyType, "inlineme", inline=costmodel)
+        Dummy, DummyType = self.make_dummy_type()
+
+        @overload_attribute(DummyType, "inlineme", inline=costmodel)
         def _get_inlineme(obj):
             def get(obj):
                 return "MY INLINED ATTRS"
@@ -1147,7 +1155,7 @@ class TestOverloadMethsAttrsInlining(InliningBase):
 
         self.check_getattr(
             test_impl=foo,
-            args=[self.Dummy()],
+            args=[Dummy()],
             expected="MY INLINED ATTRS",
             block_count=1,
             expects_inlined=should_inline,

